### PR TITLE
Use container for generating docs

### PR DESCRIPTION
**Pull request by pschmitt**
So here we go. It's all using the zinit container.

Refs:
- #185
- #196

### CODE REVIEW

File: .github/workflows/documentation.yaml [hunk 1]
```diff
@@ -28,7 +20,10 @@ jobs:
 
       - name: 👷 Generate zsd
         run: |
-          make doc
+          make doc-container
+          rc="$?"
+          sudo chown -R "$(id -u):$(id -g)" .
```
Comment (vladdoster) [inline .github/workflows/documentation.yaml:25]: Why is this needed?

File: scripts/docker-run.sh [hunk 2]
```diff
@@ -53,6 +55,16 @@ run() {
 
   local -a args=(--rm)
 
+  local cruntime=docker
+  local sudo_cmd
+  if [[ -z "$CI" ]] && command -v podman >/dev/null 2>&1
+  then
+    cruntime=podman
+    # rootless containers are a PITA
+    # https://www.tutorialworks.com/podman-rootless-volumes/
```
Comment (vladdoster) [inline scripts/docker-run.sh:64]: Is this really needed? Do we test both runtimes? I think it adds complexity while adding no real benefit.

1. Why not use [docker w/ rootless](https://docs.docker.com/engine/security/rootless/
)?
1. Do we want to support singularity containers, too?

File: Makefile [hunk 3]
```diff
@@ -20,6 +20,9 @@ testD:
 testE:
 	make VERBOSE=$(VERBOSE) NODIFF=$(NODIFF) DEBUG=$(DEBUG) OPTDUMP=$(OPTDUMP) OPTS="ignoreclosebraces" -C test test
 
+doc-container: zinit.zsh zinit-side.zsh zinit-install.zsh zinit-autoload.zsh
```
Comment (vladdoster) [inline Makefile:23]: Can we extract: 

```
zinit.zsh zinit-side.zsh zinit-install.zsh zinit-autoload.zsh
```

into variable to avoid duplicating it in `doc` target?

File: .github/workflows/documentation.yaml [hunk 4]
```diff
@@ -28,7 +20,10 @@ jobs:
 
       - name: 👷 Generate zsd
         run: |
-          make doc
+          make doc-container
+          rc="$?"
```
Comment (vladdoster) [inline .github/workflows/documentation.yaml:24]: Won't GH actions exit on their own if the `1` exit code is returned? This seems redundant

File: .github/workflows/documentation.yaml (hunk 1 above)
Comment (pschmitt) [reply .github/workflows/documentation.yaml:25]: There's an explicit chown to the user running in the container in the command run by the container. This is resetting the perms afterwards, w/o it the PDF step fails for example.

File: scripts/docker-run.sh (hunk 2 above)
Comment (pschmitt) [reply scripts/docker-run.sh:64]: What's your point? "Please remove this feature"?

The cli of both docker and podman are almost 100% the same. Besides using a different command name there is pretty much no added complexity here.

I want to support both, cause I don't want contributors to go through the hassle of installing podman or docker even though their distro ships the other.

File: Makefile (hunk 3 above)
Comment (pschmitt) [reply Makefile:23]: Maybe? I don't know how :D

File: .github/workflows/documentation.yaml (hunk 4 above)
Comment (pschmitt) [reply .github/workflows/documentation.yaml:24]: Right, it does. Will remove this.

File: scripts/docker-run.sh (hunk 2 above)
Comment (vladdoster) [reply scripts/docker-run.sh:64]: Your comment on line 63 made me ask the question

> # rootless containers are a PITA

Additionally, Podman presently only supports running containers on Linux. However, we are building a remote client which can run on Windows and macOS and manage Podman containers on a Linux system via the REST API using SSH tunneling.

Feels like a solution to a problem no one has encountered (e.g., pre-optimization).

File: Makefile (hunk 3 above)
Comment (vladdoster) [reply Makefile:23]: No worries, just a nit :^)

File: scripts/docker-run.sh (hunk 2 above)
Comment (pschmitt) [reply scripts/docker-run.sh:64]: Well, well, well. This all started cause I didn't want to:

1. install docker 
2. Argue with people about container runtimes

Sigh.

File: .github/workflows/documentation.yaml (hunk 4 above)
Comment (pschmitt) [reply .github/workflows/documentation.yaml]: Done.

[78c898e](https://github.com/zdharma-continuum/zinit/pull/200/commits/78c898ea775191714c22155522da6054aa034a3e)

**Review by alichtman — APPROVED** [review]
